pennylane/interfaces/tf.py

```diff
@@ -138,34 +138,74 @@ def _execute(self, params, **input_kwargs):
         res = self.execute_device(args, input_kwargs["device"])
         self.set_parameters(all_params, trainable_only=False)
 
-        def grad(grad_output, **tfkwargs):
-            variables = tfkwargs.get("variables", None)
+        saved_grad_matrices = {}
+
+        def _evaluate_grad_matrix(grad_matrix_fn):

```

Comment:
Maybe worth adding the "Args:" part of the docstring to mention that `grad_matrix_fn` is a string

Could also mention that we expect it to be, e.g., "jacobian" or "hessian".